src/main/scala/scraml/libs/CatsShowSupport.scala

```diff
@@ -2,37 +2,37 @@ package scraml.libs
 
 import scraml.LibrarySupport.appendObjectStats
 import scraml.{DefnWithCompanion, LibrarySupport, ModelGenContext}
-
 import scala.meta._
 
 object CatsShowSupport extends LibrarySupport {
-  private def showStats(typeName: String): List[Stat] =
+  private def showStats(classDef: Defn.Class): List[Stat] =
     q"""
       import cats.Show
-      implicit val ${Pat.Var(Term.Name(typeName + "Show"))}: Show[${Type
-      .Name(typeName)}] = Show.show {
+      implicit val ${Pat.Var(Term.Name(classDef.name.value + "Show"))}: Show[${classDef.name}] = Show.show {
         instance =>
-          val buffer = new StringBuilder($typeName)
+          val buffer = new StringBuilder(${classDef.name.value})
           buffer.append(':')
           buffer.append('\n')
 
-          0.until(instance.productArity).zip(instance.productElementNames).foreach {
-            case (index, name) =>
-              buffer.append('\t')
-              buffer.append(name)
-              buffer.append(": ")
-              buffer.append(instance.productElement(index))
-              buffer.append('\n')
+          ..${
```

Comment:
So this is how you get stats into `q`: :bulb: :+1:

Reply:
Yeah, that's a pretty cool technique for splicing in `Stat`s (a la the template pattern).  Another thing to note is by having a `List` of `Stat`s instead of a multi-statement one allows scalameta to incorporate them without creating a `Block`.  Learning this trick allows the code to be plugged in without having undesired scoping forced into code generation.